File: recursion.nim
# recursion.nim
proc factorial(n: int): int =
    if n <= 1:
        return 1
    else:
        return n * factorial(n - 1)

echo "Enter an integer: "
let input = readLine(stdin)
let param = parseInt(input)
if param == 0 and input != "0":
    echo "Invalid input"
else:
    echo "Factorial: ", factorial(param)

File: recursion.pl
# recursion.pl
sub factorial {
    my ($param) = @_;
    if ($param <= 1) {
        return 1;
    } else {
        return $param * factorial($param - 1);
    }
}

print "Enter an integer: ";
my $input = <STDIN>;
chomp($input);
my $param = int($input);

if ($param == 0 && $input ne "0") {
    print "Invalid input\n";
} else {
    print factorial($param) . "\n";
}
